builtin/logical/pki/cert_util.go

```diff
@@ -162,6 +241,291 @@ func validateCommonNames(req *logical.Request, commonNames []string, role *roleE
 	return "", nil
 }
 
+func generateCert(b *backend,
+	role *roleEntry,
+	signingBundle *caInfoBundle,
+	isCA bool,
+	req *logical.Request,
+	data *framework.FieldData) (*certutil.ParsedCertBundle, error) {
+
+	creationBundle, err := generateCreationBundle(b, role, signingBundle, nil, req, data)
+	if err != nil {
+		return nil, err
+	}
+
+	if isCA {
+		creationBundle.IsCA = isCA
+	}
+
+	parsedBundle, err := createCertificate(creationBundle)
+	if err != nil {
+		return nil, err
+	}
+
+	return parsedBundle, nil
+}
+
+// N.B.: This is only meant to be used for generating intermediate CAs.
+// It skips some sanity checks.
+func generateCSR(b *backend,
+	role *roleEntry,
+	signingBundle *caInfoBundle,
+	req *logical.Request,
+	data *framework.FieldData) (*certutil.ParsedCSRBundle, error) {
+
+	creationBundle, err := generateCreationBundle(b, role, signingBundle, nil, req, data)
+	if err != nil {
+		return nil, err
+	}
+
+	parsedBundle, err := createCSR(creationBundle)
+	if err != nil {
+		return nil, err
+	}
+
+	return parsedBundle, nil
+}
+
+func signCert(b *backend,
```

Comment:
It would be good to add more comments to all the methods in this file given how widely they are used